Assets/MixedRealityToolkit.Providers/WindowsMixedReality/XRSDK/Controllers/WindowsMixedRealityXRSDKArticulatedHand.cs

```diff
@@ -0,0 +1,333 @@
+                            System.Numerics.Vector3 scale;
+                            System.Numerics.Quaternion rotation;
+                            System.Numerics.Vector3 translation;
+                            System.Numerics.Matrix4x4.Decompose(meshTransform.Value, out scale, out rotation, out translation);
+
+                            var handMeshVertices = new Vector3[handMeshObserver.VertexCount];
+                            var handMeshNormals = new Vector3[handMeshObserver.VertexCount];
+
+                            for (int i = 0; i < handMeshObserver.VertexCount; i++)
+                            {
+                                handMeshVertices[i] = vertexAndNormals[i].Position.ToUnityVector3();
+                                handMeshNormals[i] = vertexAndNormals[i].Normal.ToUnityVector3();
+                            }
+
+                            HandMeshInfo handMeshInfo = new HandMeshInfo
+                            {
+                                vertices = handMeshVertices,
+                                normals = handMeshNormals,
+                                triangles = handMeshTriangleIndices,
+                                uvs = handMeshUVs,
+                                position = translation.ToUnityVector3(),
+                                rotation = rotation.ToUnityQuaternion()
+                            };
+
+                            CoreServices.InputSystem?.RaiseHandMeshUpdated(InputSource, ControllerHandedness, handMeshInfo);
+                        }
+                    }
+                }
+            }
+#endif // WINDOWS_UWP
+        }
+
+        private void UpdateIndexFingerData(MixedRealityInteractionMapping interactionMapping)
+        {
+            // Update the interaction data source
+            interactionMapping.PoseData = unityJointPoses[TrackedHandJoint.IndexTip];
+
+            // If our value changed raise it.
+            if (interactionMapping.Changed)
+            {
+                // Raise input system event if it's enabled
+                CoreServices.InputSystem?.RaisePoseInputChanged(InputSource, ControllerHandedness, interactionMapping.MixedRealityInputAction, interactionMapping.PoseData);
+            }
+        }
+
+        private TrackedHandJoint ConvertToTrackedHandJoint(HandFinger finger, int index)
+        {
+            switch (finger)
+            {
+                case HandFinger.Thumb: return (index == 0) ? TrackedHandJoint.Wrist : TrackedHandJoint.ThumbMetacarpalJoint + index - 1;
+                case HandFinger.Index: return TrackedHandJoint.IndexMetacarpal + index;
+                case HandFinger.Middle: return TrackedHandJoint.MiddleMetacarpal + index;
+                case HandFinger.Ring: return TrackedHandJoint.RingMetacarpal + index;
+                case HandFinger.Pinky: return TrackedHandJoint.PinkyMetacarpal + index;
+                default: return TrackedHandJoint.None;
+            }
+        }
```

Comment:
I remember that this specific area was kinda tricky because of how bones/fingers are different from the joint representations that we've adopted for MRTK - can you add a comment on the 'magic' that we're doing here (i.e. the offsetting) and how the mapping is done?

Reply:
Added a summary with more information